tests/virt_autotest/virt_utils.pm

```diff
@@ -48,14 +48,26 @@ sub repl_repo_in_sourcefile {
     my $verorig = "source.http.sles-" . get_version_for_daily_build_guest . "-64";
     my $veritem = check_var('ARCH', 'x86_64') ? $verorig : get_required_var('ARCH') . ".$verorig";
     if (get_var("REPO_0")) {
-        my $location = &virt_autotest_base::execute_script_run("", "perl /usr/share/qa/tools/location_detect_impl.pl", 60);
-        $location =~ s/[\r\n]+$//;
+        my $location = "";
+        if (check_var('ARCH', 's390x')) {
+            my $location = type_string("perl /usr/share/qa/tools/location_detect_impl.pl\n");
```

Comment:
This variable definition looks useless.

Reply:
defined $soucefile used this variable definition - $location
//
 59         my $soucefile = "/usr/share/qa/virtautolib/data/" . "sources." . "$location";
//

Reply:
I think `$location` from
```perl
my $soucefile = "/usr/share/qa/virtautolib/data/" . "sources." . "$location";
```
will be always `""` as `$location` is re-defined in the if-else clause.

Run this code locally and whatever you set `$trigger` to, result is always: `location=<123>`

```perl
use warnings;
use strict;

my $trigger = 1;
my $location = '123';
if ($trigger) {
  my $location = '456';
}
else {
  my $location = '789';
}
print "location=<$location>\n";
```

Then remove the `my` from if-else clause and the result is as expected.

Reply:
Cool, I am working on this as your mentioned.